### DISCUSSION TITLE
Service Discovery Mechanism vs ELB static list approach

### DISCUSSION THREAD

**Discussion by sbilello** [nautilus/gateway#135] [Q&A] [answered]
Hello!

Is there any support for a graphql service discovery via consul or right now it would be easy to add an ELB for each graphql service?

I am wondering why there is no schema registry in this architecture

Thanks

**Comment by prokaktus** [comment, marked as answer]
In our setup we use `IntrospectRemoteSchemas` for Graphql services inside k8s cluster and reload Gateway pods on every service deployment via k8s hooks. It gracefully reload pods and they start listening new schema. Maybe it'll help.

**Reply by sbilello** [reply]
This is for the service discovery but you don't need any schema registry? Is there any example available or reference to your setup?

**Reply by prokaktus** [reply]
Right now we don't use any. Unfortunately, I have no available examples 😞
